lib/integrations/slack/send_on_slack_service.rb

```diff
@@ -40,18 +40,24 @@ def avatar_url(sender)
     sender.try(:avatar_url) || "#{ENV['FRONTEND_URL']}/admin/avatar_square.png"
   end
 
-  def send_message
+  def send_message # rubocop:disable Metrics/AbcSize
     sender = message.sender
     sender_type = sender.class == Contact ? 'Contact' : 'Agent'
     sender_name = sender.try(:name) ? "#{sender_type}: #{sender.try(:name)}" : sender_type
-
-    @slack_message = slack_client.chat_postMessage(
-      channel: hook.reference_id,
-      text: message_content,
-      username: sender_name,
-      thread_ts: conversation.identifier,
-      icon_url: avatar_url(sender)
-    )
+    begin
+      @slack_message = slack_client.chat_postMessage(
+        channel: hook.reference_id,
+        text: message_content,
+        username: sender_name,
+        thread_ts: conversation.identifier,
+        icon_url: avatar_url(sender)
+      )
+    rescue Slack::Web::Api::Errors::AccountInactive => e
+      Rails.logger.info e
+      hook.authorization_error!
+      hook.disable if hook.enabled?
+      AdministratorNotifications::ChannelNotificationsMailer.slack_disconnect(message.account)&.deliver_later
```

Comment:
let's move the mailer into `Reauthorizable` `prompt_reauthorize` so that it gets triggered only after the error threshold.